Add vision-language model pretraining strategies blog

alaradirik opened the pull request:
This blog is meant to give a non-exhaustive overview of vision-language models and pre-training strategies.

Sections:
1. Introduction
2. Learning Strategies
    1. Contrastive Learning
    2. PrefixLM
    3. Multimodal Fusing with Cross Attention
    4. MLM / ITM
    5. No Training
3. Datasets
4. Supporting Vision-Language Models in 🤗 Transformers
5. Emerging Areas of Research
6. Conclusion

- [x] Add thumbnail

sayakpaul commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,390 @@
+---
+title: "A Dive into Pretraining Strategies for Vision-Language Models"
```
Maybe just Vision-Language Models?

alaradirik replied:
Makes sense, how about "A Deep into for Vision-Language Models"?

sayakpaul replied:
"Deep into" seems incomplete to me.

sayakpaul replied:
Or maybe just "A Dive into Vision-Language Models"?

alaradirik replied:
Sorry, I meant "A Deep Dive into Vision-Language Models" but "A Dive into Vision-Language Models" also works

sayakpaul replied:
Alright! Let's update with the latter one the.  I am outside currently, if you can, could you do it if possible?

sayakpaul commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,390 @@
+<h1>A Dive into Pretraining Strategies for Vision-Language Models</h1>
+
+<div class="blog-metadata">
+    <small>Published January 27, 2023.</small>
```
What about publishing it later next week? We have an upcoming blog on "CV at 🤗" planned.

alaradirik replied:
Makes sense, would February 2nd or 3rd work?

sayakpaul replied:
Let's do it on 3rd?

sayakpaul commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,390 @@
+
+# Table of contents
+
+1. [Vision-Language Models: Introduction]
+2. [Vision-Language Models: Learning Strategies]
+    1. [Contrastive Learning]
+    2. [PrefixLM]
+    3. [Multimodal Fusing with Cross Attention]
+    4. [MLM / ITM]
+    5. [No Training]
+3. [Vision-Language Models: Datasets]
+4. [Supporting Vision-Language Models in 🤗 Transformers]
+5. [Emerging Areas of Research]
+6. [Conclusion]
```
Update with section hyperlinks?

alaradirik replied:
Yes, updating it :)

sayakpaul commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,390 @@
+-   Detection of hate speech from social media content involving both images and text modalities.
+
+
+# Vision-Language Models: Learning Strategies
```
Leave a gap between the header and the text?

sayakpaul commented on vision_language_pretraining.md:
````diff
@@ -0,0 +1,390 @@
+
+Let’s start with ViLT and download a model pre-trained on the VQA dataset. We can do this by simply initializing the corresponding model class and calling the `from_pretrained()` method to download our desired checkpoint.
+
+```py
+
+from transformers import ViltProcessor, ViltForQuestionAnswering
+
+model = ViltForQuestionAnswering.from_pretrained("dandelin/vilt-b32-finetuned-vqa")
+
+```
````
Can we remove the unnecessary spaces?

sayakpaul commented on vision_language_pretraining.md:
````diff
@@ -0,0 +1,390 @@
+
+Next, we will download a random image of two cats and preprocess both the image and our  query question to transform them to the input format expected by the model. To do this, we can conveniently use the corresponding preprocessor class (`ViltProcessor`) and initialize it with the pre-processing configuration of the corresponding checkpoint. 
+
+```py
+
+import requests
+
+from PIL import Image
+
+processor = ViltProcessor.from_pretrained("dandelin/vilt-b32-finetuned-vqa")
+
+# download an input image
+
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+
+image = Image.open(requests.get(url, stream=True).raw)
+
+text = "How many cats are there?"
+
+# prepare inputs
+
+inputs = processor(image, text, return_tensors="pt")
+
+```
````
Can we remove the unnecessary spaces

sayakpaul commented on vision_language_pretraining.md:
````diff
@@ -0,0 +1,390 @@
+
+Finally, we can perform inference using the pre-processed image and question as input and print the predicted answer. However, an important point to keep in mind is to make sure your text input resembles the question templates used in the training setup. You can refer to [the paper and the dataset](https://arxiv.org/abs/2102.03334) to learn how the questions are formed. 
+
+```py
+
+import torch
+
+# forward pass
+
+with torch.no_grad():
+
+    outputs = model(**inputs)
+
+logits = outputs.logits
+
+idx = logits.argmax(-1).item()
+
+print("Predicted answer:", model.config.id2label[idx])
+
+```
````
Can we remove the unnecessary spaces?

sayakpaul commented on vision_language_pretraining.md:
````diff
@@ -0,0 +1,390 @@
+
+We will start by initializing `CLIPSegForImageSegmentation` and its corresponding pre-processing class and load our pre-trained model.
+
+```py
+
+from transformers import CLIPSegProcessor, CLIPSegForImageSegmentation
+
+processor = CLIPSegProcessor.from_pretrained("CIDAS/clipseg-rd64-refined")
+
+model = CLIPSegForImageSegmentation.from_pretrained("CIDAS/clipseg-rd64-refined")
+
+```
````
Can we remove the unnecessary spaces?

sayakpaul commented on vision_language_pretraining.md:
````diff
@@ -0,0 +1,390 @@
+
+Next, we will use the same input image and query the model with the text descriptions of all objects we want to segment. Similar to other pre-processors, `CLIPSegProcessor` transforms the inputs to the format expected by the model. As we want to segment multiple objects, we input the same image for each text description separately.
+
+```py
+
+from PIL import Image
+
+import requests
+
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+
+image = Image.open(requests.get(url, stream=True).raw)
+
+texts = ["a cat", "a remote", "a blanket"]
+
+inputs = processor(text=texts, images=[image] * len(texts), padding=True, return_tensors="pt")
+
+```
````
Can we remove the unnecessary spaces?

sayakpaul commented on vision_language_pretraining.md:
````diff
@@ -0,0 +1,390 @@
+
+Similar to ViLT, it’s important to refer to the [original work](https://arxiv.org/abs/2112.10003) to see what kind of text prompts are used to train the model in order to get the best performance during inference. While CLIPSeg is trained on simple object descriptions (e.g., “a car”), its CLIP backbone is pre-trained on engineered text templates (e.g., “an image of a car”, “a photo of a car”) and ...
+
+```py
+
+import torch
+
+with torch.no_grad():
+
+    outputs = model(**inputs)
+
+logits = outputs.logits
+
+print(logits.shape)
+
+>>> torch.Size([3, 352, 352])
+
+```
````
Same.

sayakpaul commented on vision_language_pretraining.md:
````diff
@@ -0,0 +1,390 @@
+
+Let’s visualize the results to see how well CLIPSeg performed (code is adapted from [this post](https://huggingface.co/blog/clipseg-zero-shot)).
+
+```py
+
+import matplotlib.pyplot as plt
+
+logits = logits.unsqueeze(1)
+
+_, ax = plt.subplots(1, len(texts) + 1, figsize=(3*(len(texts) + 1), 12))
+
+[a.axis('off') for a in ax.flatten()]
+
+ax[0].imshow(image)
+
+[ax[i+1].imshow(torch.sigmoid(logits[i][0])) for i in range(len(texts))];
+
+[ax[i+1].text(0, -15, prompt) for i, prompt in enumerate(texts)]
+
+```
````
Same.

sayakpaul commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,390 @@
+
+There have been incredible advances in multimodal models in recent years, with vision-language models making the biggest leap both in terms of performance and the variety of use cases and applications. In this blog, we talked about the latest advancements in vision-language models, as well as what multimodal datasets are available and which pre-training strategies we can use to train and fine-tun ...
+
+We are continuing to integrate the most impactful computer vision and multimodal models and would love to hear back from you. To stay up to date with the latest news in multimodal research, you can follow us on Twitter: [@adirik](https://twitter.com/adirik), [@NielsRogge](https://twitter.com/NielsRogge), [@apsdehal](https://twitter.com/apsdehal), [@a_e_roberts](https://twitter.com/a_e_roberts),   ...
+
```
Looks like a "[" is missing from my username.

sayakpaul reviewed (approved):
Thanks a lot! Left a few minor nits. 

Also, did you preview the blog using [codespaces](https://github.com/huggingface/moon-landing#codespace)? Also, we need to update the `_blog.yml`. 

@apsdehal @amyeroberts FYI. 

Also Cc: @NielsRogge @patil-suraj @younesbelkada

sayakpaul commented:
@alaradirik I had to update an entry for the blog in `_blog.yml` to preview using Codespaces. 

Some things that came up:

1) The hyperlinks are not rendering in the image captions. 

![Screenshot 2023-01-28 at 7 57 00 AM](https://user-images.githubusercontent.com/22957388/215237499-9edffc7d-a9c8-47a1-8933-e71caa97a3c6.png)

2) The hyperlinks are not rendering here: 

![Screenshot 2023-01-28 at 7 56 03 AM](https://user-images.githubusercontent.com/22957388/215237532-a2ef4080-87f6-41c3-9ba9-ee10f33ff757.png)

younesbelkada reviewed (approved):
Amazing blogpost 🔥 ! 
BLIP also performs ITM, but not sure about its training strategy so I am not sure if it's relevant to add it in `4) Masked-Language Modeling / Image-Text Matching` wdyt?

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+
+A vision-language model typically consists of 3 key elements: an image encoder, a text encoder, and a strategy to fuse information from the two encoders. These key elements are tightly coupled together as the loss functions are designed around both the model architecture and the learning strategy. While vision-language model research is hardly a new research area, the design of such models has ch ...
+
+In this section, we’ll discuss some of the typical pre-training objectives and strategies for vision-language models that have been shown to perform well as far as their transfer performance is concerned. We’ll also touch upon additional interesting things that are either specific to these objectives or can be used as general components for pre-training. 
```
```suggestion
In this section, we'll discuss some of the typical pre-training objectives and strategies for vision-language models that have been shown to perform well regarding their transfer performance. We'll also touch upon additional interesting things that are either specific to these objectives or can be used as general components for pre-training. 
```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+
+Pre-trained vision-language models are often trained on various downstream tasks such as visual question-answering, text-guided object detection, text-guided image inpainting, multi-modal classification, and various stand-alone NLP and computer vision tasks. 
+
+Models finetuned on the question-answering downstream task, such as [ViLT](https://arxiv.org/abs/2102.03334) and [GLIP](https://arxiv.org/abs/2112.03857), most commonly use the [VQA](https://visualqa.org/) (visual question-answering), [VQA v2](https://visualqa.org/), [NLVR2](https://lil.nlp.cornell.edu/nlvr/), [OKVQA](https://okvqa.allenai.org/), [TextVQA](https://huggingface.co/datasets/textvqa) ...
```
```suggestion
Models fine-tuned on the question-answering downstream task, such as [ViLT](https://arxiv.org/abs/2102.03334) and [GLIP](https://arxiv.org/abs/2112.03857), most commonly use the [VQA](https://visualqa.org/) (visual question-answering), [VQA v2](https://visualqa.org/), [NLVR2](https://lil.nlp.cornell.edu/nlvr/), [OKVQA](https://okvqa.allenai.org/), [TextVQA](https://huggingface.co/datasets/textvqa), [TextCaps](https://textvqa.org/textcaps/) and [VizWiz](https://vizwiz.org/) datasets. These datasets typically contain images paired with multiple open-ended questions and answers. Furthermore, datasets such as VizWiz and TextCaps can also be used for image segmentation and object localization downstream tasks. Some other interesting multi-modal downstream datasets are [Hateful Memes](https://huggingface.co/datasets/limjiayi/hateful_memes_expanded) for multi-modal classification, [SNLI-VE](https://github.com/necla-ml/SNLI-VE) for visual entailment prediction, and [Winoground](https://huggingface.co/datasets/facebook/winoground) for visio-linguistic compositional reasoning. 
```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+
+While fusing visual information into a language model is highly effective, being able to use a pre-trained language model (LM) without the need for fine-tuning would be much more efficient. Hence, another pre-training objective in vision-language models is learning image embeddings that are aligned with a frozen language model. 
+
+Models such as [Frozen](https://arxiv.org/abs/2106.13884), [MAPL, ](https://arxiv.org/abs/2210.07179)and [ClipCap](https://arxiv.org/abs/2111.09734) use this Frozen PrefixLM pre-training objective. They only update the parameters of the image encoder during training to generate image embeddings that can be used as a prefix to the pre-trained, frozen language model, in a similar fashion to the Pre ...
```
```suggestion
Models such as [Frozen](https://arxiv.org/abs/2106.13884), [MAPL, ](https://arxiv.org/abs/2210.07179)and [ClipCap](https://arxiv.org/abs/2111.09734) use this Frozen PrefixLM pre-training objective. They only update the parameters of the image encoder during training to generate image embeddings that can be used as a prefix to the pre-trained, frozen language model in a similar fashion to the PrefixLM objective discussed above. Both Frozen and ClipCap are trained on aligned image-text (caption) datasets with the objective of generating the next token in the caption, given the image embeddings and the prefix text. 
```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+
+Another approach to leveraging pre-trained language models for multimodal tasks is to directly fuse visual information into the layers of a language model decoder using a cross-attention mechanism instead of using images as additional prefixes to the language model. Models such as [VisualGPT](https://arxiv.org/abs/2102.10407), [VC-GPT](https://arxiv.org/abs/2201.12723), and [Flamingo](https://arx ...
+
+Models such as VisualGPT use a visual encoder to embed images and feed the visual embeddings to the cross-attention layers of a pre-trained language decoder module to generate plausible captions. A more recent work, [FIBER](http://arxiv.org/abs/2206.07643), inserts cross-attention layers with a gating mechanism into both vision and language backbones for more efficient multimodal fusing and enabl ...
```
```suggestion
Models such as VisualGPT use a visual encoder to embed images and feed the visual embeddings to the cross-attention layers of a pre-trained language decoder module to generate plausible captions. A more recent work, [FIBER](http://arxiv.org/abs/2206.07643), inserts cross-attention layers with a gating mechanism into both vision and language backbones, for more efficient multimodal fusing and enables various other downstream tasks, such as image-text retrieval and open vocabulary object detection. 

```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+Models such as VisualGPT use a visual encoder to embed images and feed the visual embeddings to the cross-attention layers of a pre-trained language decoder module to generate plausible captions. A more recent work, [FIBER](http://arxiv.org/abs/2206.07643), inserts cross-attention layers with a gating mechanism into both vision and language backbones for more efficient multimodal fusing and enabl ...
+### 4) Masked-Language Modeling / Image-Text Matching
+
+Another line of vision-language models use a combination of Masked-Language Modeling (MLM) and Image-Text Matching (ITM) objectives to align specific parts of images with text and enable various downstream tasks such as visual question answering, visual commonsense reasoning, text-based image retrieval and text-guided object detection. Models that follow this pre-training setup include  [VisualBE ...
```
```suggestion
Another line of vision-language models uses a combination of Masked-Language Modeling (MLM) and Image-Text Matching (ITM) objectives to align specific parts of images with text and enable various downstream tasks such as visual question answering, visual commonsense reasoning, text-based image retrieval, and text-guided object detection. Models that follow this pre-training setup include [VisualBERT](https://arxiv.org/abs/1908.03557), [FLAVA](https://arxiv.org/abs/2112.04482), [ViLBERT](https://arxiv.org/abs/1908.02265) and [LXMERT](https://arxiv.org/abs/1908.07490).
```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+Another work, FLAVA, consists of an image encoder, a text encoder, and a multimodal encoder to fuse and align the image and text representations for multimodal reasoning, all of which are based on transformers. In order to achieve this, FLAVA uses a variety of pre-training objectives: MLM, ITM, as well as Masked-Image Modeling (MIM), and contrastive learning.
+### 5) No Training 
+
+Finally, there are various optimization strategies that aim to bridge image and text representations using the pre-trained image and text models or adapt pre-trained multimodal models to new downstream tasks without any additional training. 
```
```suggestion
Finally, various optimization strategies aim to bridge image and text representations using the pre-trained image and text models or adapt pre-trained multimodal models to new downstream tasks without additional training. 
```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+
+## Vision-Language Models: Datasets
+
+Vision-language models are typically trained on large image and text datasets with different structures based on the pre-training objective. After they are pre-trained, they are further fine-tuned on various downstream tasks using task-specific datasets. In this section, we provide an overview of some popular pre-training and downstream datasets used for training and evaluating vision-language mo ...
```
```suggestion
Vision-language models are typically trained on large image and text datasets with different structures based on the pre-training objective. After they are pre-trained, they are further fine-tuned on various downstream tasks using task-specific datasets. This section provides an overview of some popular pre-training and downstream datasets used for training and evaluating vision-language models.

```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+
+Vision-language models are typically pre-trained on large multi-modal datasets harvested from the web in the form of matching image/video and text pairs. The text data in these datasets can be human-generated captions, automatically generated captions, image metadata, or simple object labels. Some examples of such large datasets are [PMD](https://huggingface.co/datasets/facebook/pmd) and [LAION-5 ...
+
+Note that even image-text datasets that solely consist of human-generated captions such as Flickr30K are inherently noisy as users do not always write descriptive or reflective captions for their images. To overcome this issue, datasets such as the LAION-5B dataset leverage CLIP or other pre-trained multimodal models to filter noisy data and create high-quality multimodal datasets. Furthermore, s ...
```
```suggestion
Even image-text datasets consisting solely of human-generated captions, such as Flickr30K, are inherently noisy as users only sometimes write descriptive or reflective captions for their images. To overcome this issue, datasets such as the LAION-5B dataset leverage CLIP or other pre-trained multimodal models to filter noisy data and create high-quality multimodal datasets. Furthermore, some vision-language models, such as ALIGN, propose further preprocessing steps and create their own high-quality datasets. Other vision-language datasets, such as the [LSVTD](https://davar-lab.github.io/dataset/lsvtd.html) and [WebVid](https://github.com/m-bain/webvid) datasets, consist of video and text modalities, although at a smaller scale.

```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+
+Models finetuned on the question-answering downstream task, such as [ViLT](https://arxiv.org/abs/2102.03334) and [GLIP](https://arxiv.org/abs/2112.03857), most commonly use the [VQA](https://visualqa.org/) (visual question-answering), [VQA v2](https://visualqa.org/), [NLVR2](https://lil.nlp.cornell.edu/nlvr/), [OKVQA](https://okvqa.allenai.org/), [TextVQA](https://huggingface.co/datasets/textvqa) ...
+
+Note that vision-language models are used for various classical NLP and computer vision tasks such as text or image classification, and typically use uni-modal datasets ([SST2](https://huggingface.co/datasets/sst2), [ImageNet-1k](https://huggingface.co/datasets/imagenet-1k), for example) for such downstream tasks. In addition, datasets such as [COCO](https://cocodataset.org/), and [Conceptual Cap ...
```
```suggestion
Note that vision-language models are used for various classical NLP and computer vision tasks such as text or image classification and typically use uni-modal datasets ([SST2](https://huggingface.co/datasets/sst2), [ImageNet-1k](https://huggingface.co/datasets/imagenet-1k), for example) for such downstream tasks. In addition, datasets such as [COCO](https://cocodataset.org/) and [Conceptual Captions](https://ai.google.com/research/ConceptualCaptions/) are commonly used both in the pre-training of models and also for the caption generation downstream task. 

```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+
+Unlike other models, the `VisionEncoderDecoderModel` is a cookie-cutter model that can be used to initialize an image-to-text model with any pre-trained Transformer-based vision model as the encoder (e.g. ViT, BEiT, DeiT, Swin) and any pre-trained language model as the decoder (e.g. RoBERTa, GPT2, BERT, DistilBERT). In fact, TrOCR is an instance of this cookie-cutter class.
+
+Let’s go ahead and experiment with some of these models. We will use [ViLT](https://huggingface.co/docs/transformers/model_doc/vilt) for visual question answering and [CLIPSeg](https://huggingface.co/docs/transformers/model_doc/clipseg) for zero-shot image segmentation. First, let’s install 🤗Transformers: `pip install transformers`.
```
```suggestion
Let’s go ahead and experiment with some of these models. We will use [ViLT](https://huggingface.co/docs/transformers/model_doc/vilt) for visual question answering and [CLIPSeg](https://huggingface.co/docs/transformers/model_doc/clipseg) for zero-shot image segmentation. First, let’s install 🤗Transformers: `pip install transformers`.

```

osanseviero commented on vision_language_pretraining.md:
````diff
@@ -0,0 +1,326 @@
+print("Predicted answer:", model.config.id2label[idx])
+```
+
+Straight-forward, right? Let’s do another demonstration with CLIPSeg and see how we can perform zero-shot image segmentation with a few lines of code. 
+### CLIPSeg for zero-shot image segmentation
````
```suggestion
Straight-forward, right? Let’s do another demonstration with CLIPSeg and see how we can perform zero-shot image segmentation with a few lines of code. 

### CLIPSeg for zero-shot image segmentation
```

osanseviero commented on vision_language_pretraining.md:
````diff
@@ -0,0 +1,326 @@
+inputs = processor(text=texts, images=[image] * len(texts), padding=True, return_tensors="pt")
+```
+
+Similar to ViLT, it’s important to refer to the [original work](https://arxiv.org/abs/2112.10003) to see what kind of text prompts are used to train the model in order to get the best performance during inference. While CLIPSeg is trained on simple object descriptions (e.g., “a car”), its CLIP backbone is pre-trained on engineered text templates (e.g., “an image of a car”, “a photo of a car”) and ...
````
Sometimes we write preprocess and sometimes pre-process. Let's try to be consistent please

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+While robotics research hasn’t leveraged vision-language models on a wide scale yet, we see works such as [CLIPort](https://arxiv.org/abs/2109.12098) leveraging joint vision-language representations for end-to-end imitation learning and reporting large improvements over previous SOTA. We also see that large language models are increasingly getting adopted in robotics tasks such as common sense re ...
+## Conclusion
+
+There have been incredible advances in multimodal models in recent years, with vision-language models making the biggest leap both in terms of performance and the variety of use cases and applications. In this blog, we talked about the latest advancements in vision-language models, as well as what multimodal datasets are available and which pre-training strategies we can use to train and fine-tun ...
```
```suggestion
There have been incredible advances in multimodal models in recent years, with vision-language models making the most significant leap in performance and the variety of use cases and applications. In this blog, we talked about the latest advancements in vision-language models, as well as what multimodal datasets are available and which pre-training strategies we can use to train and fine-tune such models. We also showed how these models are integrated into 🤗 Transformers and how you can use them to perform various tasks with a few lines of code. 
```

osanseviero commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+
+We are continuing to integrate the most impactful computer vision and multimodal models and would love to hear back from you. To stay up to date with the latest news in multimodal research, you can follow us on Twitter: [@adirik](https://twitter.com/https://twitter.com/alaradirik), [@NielsRogge](https://twitter.com/NielsRogge), [@apsdehal](https://twitter.com/apsdehal), [@a_e_roberts](https://twi ...
+
+*Acknowledgements: We thank Amanpreet Singh and Amy Roberts for their rigorous reviews. Also, thanks to Niels Rogge, Younes Belkada, Suraj Patil among many others at Hugging Face, who laid out the foundations for increasing the use of multimodal models from Transformers.*
```
```suggestion
*Acknowledgements: We thank Amanpreet Singh and Amy Roberts for their rigorous reviews. Also, thanks to Niels Rogge, Younes Belkada, and Suraj Patil, among many others at Hugging Face, who laid out the foundations for increasing the use of multimodal models from Transformers.*
```

sayakpaul commented on vision_language_pretraining.md:
```diff
@@ -0,0 +1,326 @@
+5. [Emerging Areas of Research](#emerging-areas-of-research)
+6. [Conclusion](#conclusion)
+
+## Vision-Language Models: Introduction
```
Good one actually. We will just keep it in "Supporting Vision-Language Models in 🤗 Transformers".

sayakpaul commented:
@younesbelkada thank you ^_^

> BLIP also performs ITM, but not sure about its training strategy so I am not sure if it's relevant to add it in 4) Masked-Language Modeling / Image-Text Matching wdyt?

Since BLIP has additional objectives too, we didn't include it. 

@osanseviero thanks for your comments and suggestions. Done addressing them all.

alaradirik commented:
> @alaradirik I had to update an entry for the blog in `_blog.yml` to preview using Codespaces.

> **Update** Both have been fixed now.

Thank you @sayakpaul, I added BridgeTower and fixed the merge conflict, I will merge the PR on Friday.

Also thanks to @osanseviero and @younesbelkada for the reviews!
